### PR TITLE
CHIA-4261: Use bisect for request_ses_hashes SES lookup

### PR DESCRIPTION
**Pull request by emlowe**
## Summary
- `request_ses_hashes` walked all SES heights linearly (~23k on mainnet) on every wallet request, which could stall the full-node event loop under a pipelined flood (SEC-1108).
- Replace that scan with `bisect_right` while keeping the same response shape. Still calls `get_ses_heights()` (sort) — this PR only addresses the larger scan cost.

## Test plan
- [x] Equivalence check vs previous linear logic on edge cases (empty match, single interval, span two SES, `uint32.MAX`)
- [x] Pi4 microbench: sort+scan vs sort+bisect under attack-shaped requests
- [x] CI

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Comment by emlowe** [comment]
close and reopen for CI

**Review by almogdepaz — DISMISSED** [review]

File: chia/full_node/full_node_api.py [hunk 1]
```diff
@@ -1952,27 +1953,24 @@ async def request_ses_hashes(self, request: wallet_protocol.RequestSESInfo) -> M
...
+                if ses_start_height <= start_height < next_ses_height:
+                    ses_hash_heights.append([ses_start_height, next_ses_height])
+                    ses: SubEpochSummary = self.full_node.blockchain.get_ses(ses_start_height)
+                    ses_reward_hashes.append(ses.reward_chain_hash)
+                    if not (ses_start_height < end_height < next_ses_height) and idx < len(ses_height) - 2:
+                        # else add extra ses as request start <-> end spans two ses
+                        next_next_height = ses_height[idx + 2]
+                        ses_hash_heights.append([next_ses_height, next_next_height])
+                        nex_ses: SubEpochSummary = self.full_node.blockchain.get_ses(next_ses_height)
+                        ses_reward_hashes.append(nex_ses.reward_chain_hash)
```
Comment (arvidn) [inline chia/full_node/full_node_api.py]: If this was broken out into a separate (free) function, I think the unit test would be a lot simpler and straight forward.

**Review by arvidn — COMMENTED** [review]
it's good to optimize handling this request. However, I don't think it's the core of the issue. I think adding an async sleep of 500ms would make sense in addition.

**Review by arvidn — APPROVED** [review]
